04_numeronym/n0npax/main_test.go

```diff
@@ -0,0 +1,54 @@
+package main
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+
+	main()
+
+	expected := strconv.Quote("[K8s anz i18n]\n")
+	actual := strconv.Quote(buf.String())
+
+	if expected != actual {
+		t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected, actual)
+	}
+}
+
+func TestNumeronyms(t *testing.T) {
+	actual := numeronyms("K8bernetes", "anz", "internationalization")
+	expected := []string{"K8s", "anz", "i18n"}
+	for i, v := range actual {
+		if expected[i] != v {
+			t.Errorf("Unexpected output. Expected: %q - Actual: %q", expected[i], v)
+		}
+	}
+}
+
+func TestShorten(t *testing.T) {
+	testCases := map[string]struct {
+		input    string
+		expected string
+	}{
+		"anz":                  {input: "anz", expected: "anz"},
+		"Kubernetes":           {input: "Kubernetes", expected: "K8s"},
+		"internationalization": {input: "internationalization", expected: "i18n"},
+	}
+	for name, test := range testCases {
+		t.Run(name, helperTestShorten(test.input, test.expected))
```

Comment:
Could we just use the anonymous function inline?  I don't believe we would lose any readability, it might make it a bit more succinct.

Reply:
I will cause referring to variables out of the scope and `golangci-lint` will complain

Reply:
I believe it is probably complaining because the function is capture the test value which will possibly be incorrect by the time the function is executed.  You can resolve this by shadowing the variable.

```go
for name, test := range cases {
		input := test.input
		want := test.want
		t.Run(name, func(t *testing.T) {
			assert.Equalf(t, want, numeronyms(input...), "%v was not sorted", input)
		})
	}
```

If you really want to use a separate function do we still need the internal anonymous function inside `helperTestShorten`?  Is it possible to use it directly?

Reply:
The suggested solution works. Thanks!
I'm surprised the shadowing solves this. `want` and `input` is still outside of the anonymous function.